app/models/institution.rb

```diff
@@ -1,4 +1,6 @@
 class Institution < ActiveRecord::Base
   validates_length_of :name, minimum: 1
   validates_uniqueness_of :name
+  # Establishing relationship between Course and Institution
+  has_many :courses, dependent: :destroy
```

Comment:
This means deleting one institution will cause a bunch of course got deleted. Does not sound like the right thing to do.